spring-test/src/main/java/org/springframework/test/context/jdbc/SqlScriptsTestExecutionListener.java

```diff
@@ -366,7 +413,11 @@ private String[] getScripts(Sql sql, Class<?> testClass, Method testMethod, bool
 	 * Detect a default SQL script by implementing the algorithm defined in
 	 * {@link Sql#scripts}.
 	 */
-	private String detectDefaultScript(Class<?> testClass, Method testMethod, boolean classLevel) {
+	private String detectDefaultScript(Class<?> testClass, @Nullable Method testMethod, boolean classLevel) {
+		if (!classLevel && testMethod == null) {
+			throw new AssertionError("Method-level @Sql requires a testMethod");
+		}
```

Comment:
We don't throw `AssertionError` in such scenarios.

Please use `Assert.state(...)` instead.